### PR TITLE
Generate changelog during release pipeline

### LINKED ISSUE #740: Run changelog generation on CI
According to the decision https://github.com/kyma-project/community/blob/master/sig-and-wg/sig-core/decisions/dr-019-Changelog_generation_with_PR_labels.md, the changelog should be generated as a part of the CI pipeline.

## To do:
- [x] Use lerna-changelog tool
- [x] Generate changelog in a form of CHANGELOG.md file
- [x] Use Github API to add content to Github release with a changelog for a specific version

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Generate CHANGELOG.md file with full changelog during release process
- Commit and push CHANGELOG.md before GitHub release
- Publish release changelog for GitHub releases during release process

**Related issue(s)**
Resolves #740

### CODE REVIEW

File: tools/changelog-generator/app/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "description": "Changelog Generator",
+  "license": "Apache-2.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/kyma-project/kyma"
+  },
+  "changelog": {
+    "repo": "kyma-project/kyma",
+    "labels": {
+      "area/application-connector": "Application Connector",
```
Comment (michal-hudy) [inline tools/changelog-generator/app/package.json]: It should be described somewhere.

File: tools/changelog-generator/app/push-full-changelog.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+echo "Test"
```
Comment (magicmatatjahu) [inline tools/changelog-generator/app/push-full-changelog.sh]: Those `echos` and `pwd` is important?

File: tools/changelog-generator/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                        stage("build image $application") {
+                            dir(env.APP_FOLDER) {
+                                sh "docker build -t ${dockerPushRoot}${application}:latest . --label version=${dockerImageTag} --label component=${application}"
+                            }
+                        }
+
+                        stage("push image $application") {
+                            sh "docker tag ${dockerPushRoot}${application}:latest ${dockerPushRoot}${application}:${dockerImageTag}"
+                            sh "docker push ${dockerPushRoot}${application}:${dockerImageTag}"
+                            if (params.GIT_BRANCH == 'master') {
```
Comment (magicmatatjahu) [inline tools/changelog-generator/Jenkinsfile]: `isMaster` global variable would be better 😄

File: tools/changelog-generator/app/generate-full-changelog.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env sh
+
+source ${APP_PATH}/variables.sh
+
+CONFIGURE_GIT=$1
+if [ "$CONFIGURE_GIT" = "--configure-git" ]; then
+    echo "Configuring git..."
+    source ${APP_PATH}/config-git.sh
+fi
```
Comment (magicmatatjahu) [inline tools/changelog-generator/app/generate-full-changelog.sh]: Splitting code to this line to separate file and use it in `tools/changelog-generator/app/generate-full-changelog.sh` and `tools/changelog-generator/app/generate-release-changelog.sh` would be great 👍

File: release.Jenkinsfile [hunk 5]
```diff
@@ -88,7 +88,11 @@ try {
                         }
                     
                         commitID = sh (script: "git rev-parse origin/${params.RELEASE_BRANCH}", returnStdout: true).trim()
-                        configureBuilds(commitID)
+                        configureBuilds()
+
+                        withCredentials([usernamePassword(credentialsId: env.DOCKER_CREDENTIALS, passwordVariable: 'pwd', usernameVariable: 'uname')]) {
```
Comment (suleymanakbas91) [inline release.Jenkinsfile]: As far as I see you're only pulling images and not pushing anything. For that, you don't need to login.

File: release.Jenkinsfile [hunk 6]
```diff
@@ -147,12 +148,30 @@ try {
...
+                            changelogGenerator('/app/generate-full-changelog.sh --configure-git', ["LATEST_VERSION=${appVersion}", "GITHUB_AUTH=${token}", "SSH_FILE=${sshfile}"])
+                            sh "BRANCH=${params.RELEASE_BRANCH} LATEST_VERSION=${appVersion} SSH_FILE=${sshfile} APP_PATH=./tools/changelog-generator/app ./tools/changelog-generator/app/push-full-changelog.sh --configure-git"
+                            commitID = sh (script: "git rev-parse HEAD", returnStdout: true).trim()
+
+                            def releaseChangelog = readFile "./.changelog/release-changelog.md"
+                            def body = releaseChangelog.replaceAll("(\\r|\\n|\\r\\n)+", "\\\\n")
+                            def data = "'{\"tag_name\": \"${appVersion}\",\"target_commitish\": \"${commitID}\",\"name\": \"${appVersion}\",\"body\": \"${body}\",\"draft\": false,\"prerelease\": ${isRelease ? 'false' : 'true'}}'"
                             def json = sh (script: "curl --data ${data} -H \"Authorization: token $token\" https://api.github.com/repos/kyma-project/kyma/releases", returnStdout: true)
+                            echo json
```
Comment (michal-hudy) [inline release.Jenkinsfile]: Do we need that `echo`?

**Comment by pkosiec** [comment]
@derberg All improvements you've mentioned are already included in this PR

**Pull request merged**